### PR TITLE
Fix memory leak in paginated tables [prod-leak-fix]

### PR DESCRIPTION
**Pull request by ValentinVignal**
Part of https://github.com/flutter/flutter/issues/141198

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

**Comment by ValentinVignal** [comment]
cc @polina-c

File: packages/flutter/test/material/paginated_data_table_test.dart [hunk 1]
```diff
@@ -70,107 +71,118 @@ void main() {
   setUp(() => source = TestDataSource());
   tearDown(() => source.dispose());
 
-  testWidgets('PaginatedDataTable paging', (WidgetTester tester) async {
-    final List<String> log = <String>[];
+  testWidgets(
```
Comment (polina-c) [inline packages/flutter/test/material/paginated_data_table_test.dart:74]: See https://github.com/flutter/flutter/pull/146748/files#r1566025452.
Can you also fix indent introduced by https://github.com/flutter/flutter/pull/146747 , here?

Comment (ValentinVignal) [reply packages/flutter/test/material/paginated_data_table_test.dart:74]: Sure, let me revert and fix that 👍

Comment (ValentinVignal) [reply packages/flutter/test/material/paginated_data_table_test.dart]: [Revert "test: Update the tests"](https://github.com/flutter/flutter/pull/146755/commits/4cce57066bfe131d6977d299acb0d6ed6b5e88eb) and [Revert bad formatting in transitions_test.dart](https://github.com/flutter/flutter/pull/146755/commits/0734f55dbdf73c5c880f6bac592e7d8e321b6296) should have taken care of that

**Review by polina-c — APPROVED** [review]

**Comment by ValentinVignal** [comment]
@polina-c I'm not sure why "Google testing" is failing. I looks like I cannot access the logs

**Comment by polina-c** [comment]
Yes, it is accessible just for googlers. I will investigate

**Comment by polina-c** [comment]
It is infrastructure error. Updated branch to restart tests.

**Comment by polina-c** [comment]
Google testing failed again. Looks like infra error that reproduces just for this PR. Working with owners of this process.

**Comment by polina-c** [comment]
Okay, dependent commits on this PR cause the mechanism to break. There are two options:
(1) redo the PR without dependent commits and (2) wait few days for concurrent PR to settle that will resolve the issue.
I am ok with both.

**Comment by ValentinVignal** [comment]
@polina-c I rebased the branch with clean commits and force-pushed it. Hopefully it will solve it 🤞

**Comment by polina-c** [comment]
Yes! Let's wait fro tree.

**Pull request merged**